### PR TITLE
Tweak Eastern Invasion difficulty

### PR DESCRIPTION
**Pull request by cooljeanius**
These are mostly just changes to gold/income/turns values, so they should be pretty safe to merge. Note that I originally developed these changes against 1.14, and @nemaara has already made some of her own changes to Eastern Invasion as part of 1.15, so I just dropped the parts that didn't apply anymore, or that didn't make sense anymore. This means I haven't actually tested these changes in this form, just in the form that appears in [https://github.com/cooljeanius/wesnoth_mods/tree/master/campaigns/Eastern_Invasion](https://github.com/cooljeanius/wesnoth_mods/tree/master/campaigns/Eastern_Invasion). They should be similar enough that that shouldn't make much of a difference, though.

### CODE REVIEW

File: data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg [hunk 1]
```diff
@@ -53,9 +53,9 @@
         {FLAG_VARIANT6 ragged}
     [/side]
 
-    {LIMIT_CONTEMPORANEOUS_RECRUITS 2 "Direwolf Rider" 3}
+    {LIMIT_CONTEMPORANEOUS_RECRUITS 2 "Direwolf Rider" {ON_DIFFICULTY 1 2 3}}
 
-    {STARTING_VILLAGES 2 8}
+    {STARTING_VILLAGES 2 {ON_DIFFICULTY 4 6 8}}
```
Comment (nemaara) [inline data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg:58]: Same thing as the other comment, I don't think varying the villages so much based on difficulty is good. If you want to keep this, I think it should be 6 7 8.

File: data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg [hunk 2]
```diff
@@ -44,7 +44,7 @@
         controller=ai
         recruit=Orcish Warrior,Goblin Knight
         {GOLD 100 150 200}
-        {INCOME 20 25 30}
+        {INCOME 10 20 30}
```
Comment (nemaara) [inline data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg:47]: This income variation is a little too much. I'm okay with something like {16 24 30} but having it be a third on easy difficulty means there'll be absolutely no pressure on you this scenario. Same applies for the other sides here.

File: data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg [hunk 3]
```diff
@@ -184,12 +184,23 @@
             id=Engineer
             name= _ "Engineer"
             unrenamable=yes
-            type=Red Mage
+            type={ON_DIFFICULTY "Arch Mage" "Silver Mage" "Red Mage"}
```
Comment (nemaara) [inline data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg:187]: I think it's better to not vary the type based on difficulty. Some people may prefer silver, some arch. (I personally almost always use silver mages)

File: data/campaigns/Eastern_Invasion/scenarios/17b_Weldyn_Besieged.cfg [hunk 4]
```diff
@@ -347,7 +349,7 @@
             id=Mal-Ravanal
             name= _ "Mal-Ravanal"
             profile=portraits/mal-ravanal.png
-            hitpoints=80
+            hitpoints=80 # Full amount; I considered having him start with less but it doesn't make sense for him to start injured
```
Comment (nemaara) [inline data/campaigns/Eastern_Invasion/scenarios/17b_Weldyn_Besieged.cfg:352]: Don't know if this comment is necessary.

File: data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg [hunk 5]
```diff
@@ -184,12 +184,23 @@
...
             x,y=18,5
             side=1
             random_traits=no
             [modifications]
                 {TRAIT_LOYAL}
+#ifdef EASY
```
Comment (nemaara) [inline data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg:193]: I'm not totally sure what to think about mr. engineer having extra hp. Maybe could be okay? Would help if you could elaborate a tiny bit on why you want this.

**Review by nemaara — CHANGES_REQUESTED** [review]

File: data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg (hunk 5 above)
Comment (cooljeanius) [reply data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg:193]: I just had a really hard time keeping him alive; I'm open to replacing it with an additional trait or two instead if that'd make more sense

File: data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg (hunk 1 above)
Comment (cooljeanius) [reply data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg]: ok, changed in 33a48a5

File: data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg (hunk 2 above)
Comment (cooljeanius) [reply data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg]: changed in 33a48a5

File: data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg (hunk 3 above)
Comment (cooljeanius) [reply data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg]: reverted in 33a48a5

File: data/campaigns/Eastern_Invasion/scenarios/17b_Weldyn_Besieged.cfg (hunk 4 above)
Comment (cooljeanius) [reply data/campaigns/Eastern_Invasion/scenarios/17b_Weldyn_Besieged.cfg]: removed in 33a48a5

File: data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg (hunk 5 above)
Comment (nemaara) [reply data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg:193]: I'd say traits seem better to me. Maybe resilient and another should you so choose. (Healthy just for fun?)

Comment (cooljeanius) [reply data/campaigns/Eastern_Invasion/scenarios/12_Evacuation.cfg:193]: ok I replaced it with those traits in b56b92c

**Review by nemaara — APPROVED** [review]

**Pull request merged**